### PR TITLE
Add few important links

### PR DESCRIPTION
**Pull request by ChetanKarwa**
I have addressed some of the feedbacks which were acknowledged by the reviewers in this [PR](https://github.com/fortran-lang/fpm/pull/402).
- [Suggestion from Laurence](https://github.com/fortran-lang/fpm/pull/402#pullrequestreview-616528386)
- [Suggestion from Sebastian](https://github.com/fortran-lang/fpm/pull/402#pullrequestreview-616539343)
- [Suggestion from Jon](https://github.com/fortran-lang/fpm/pull/402#issuecomment-802931356)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,20 +5,20 @@ Its key goal is to improve the user experience of Fortran programmers.
 It does so by making it easier to build your Fortran program or library, run the
 executables, tests, and examples, and distribute it as a dependency to other
 Fortran projects.
-Fpm's user interface is modeled after [Rust's Cargo](https://crates.io/),
+Fpm's user interface is modeled after [Rust's Cargo](https://doc.rust-lang.org/cargo/) ([Github Repo](https://github.com/rust-lang/cargo)),
```
Comment (milancurcic) [inline README.md:8]: No need to point also to the GH repo

```suggestion
Fpm's user interface is modeled after [Rust's Cargo](https://doc.rust-lang.org/cargo/),
```

File: README.md [hunk 2]
```diff
@@ -5,20 +5,20 @@ Its key goal is to improve the user experience of Fortran programmers.
...
 You can use it to build and package your Fortran projects, as well as to use
-existing fpm packages as dependencies.
+[existing fpm packages](https://github.com/fortran-lang/fpm-registry) as dependencies.
 Fpm's behavior and user interface may change as it evolves, however as fpm
 matures and we enter production, we will aim to stay backwards compatible.
 Please follow the [issues](https://github.com/fortran-lang/fpm/issues) to
 contribute and/or stay up to date with the development.
 Before opening a bug report or a feature suggestion, please read our
-[Contributor Guide](CONTRIBUTING.md).
+[Contributor Guide](CONTRIBUTING.md). You can also discuss your ideas and queries with the community on [Fortran-Lang Discourse](https://fortran-lang.discourse.group/)
```
Comment (milancurcic) [inline README.md:21]: I believe in the original suggestion, @awvwgk meant the fpm Discussions board, but good to mention both that and the Discourse.

```suggestion
[Contributor Guide](CONTRIBUTING.md). You can also discuss your ideas and queries with the community in [fpm discussions](https://github.com/fortran-lang/fpm/discussions), or more broadly on [Fortran-Lang Discourse](https://fortran-lang.discourse.group/)
```

**Review by milancurcic — CHANGES_REQUESTED** [review]
Looks good, thanks! I left two requests for changes.

**Review by LKedward — COMMENTED** [review]
Thanks @ChetanKarwa! With Milan's suggestions this looks good to me :+1:

**Comment by ChetanKarwa** [comment]
I have made the requested changes. It's good to go now.

**Review by milancurcic — APPROVED** [review]
Thank you, @ChetanKarwa

**Pull request merged**